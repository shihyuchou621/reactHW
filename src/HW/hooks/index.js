import HW1 from './HW1';
import HW2 from './HW2';
import HW3 from './HW3';
import HW4 from './HW4';
import HW5 from './HW5';
import HW6 from './HW6';
import HW7 from './HW7';
import HW8 from './HW8';
import HW9 from './HW9';
import HW10 from './HW10';
import HW11 from './HW11';
import HW12 from './HW12';
import HW13 from './HW13';
import HW14 from './HW14';
import HW15 from './HW15';
import HW16 from './HW16';
import HW17 from './HW17';
import HW18 from './HW18';
import HW19 from './HW19';
import HW20 from './HW20';
import HW21 from './HW21';
import HW22 from './HW22';
import HW23 from './HW23';
import HW25 from './HW25';
import HW26 from './HW26';
import HW27 from './HW27';

export default {
  HW1,
  HW2,
  HW3,
  HW4,
  HW5,
  HW6,
  HW7,
  HW8,
  HW9,
  HW10,
  HW11,
  HW12,
  HW13,
  HW14,
  HW15,
  HW16,
  HW17,
  HW18,
  HW19,
  HW20,
  HW21,
  HW22,
  HW23,
  HW25,
  HW26,
  HW27,
};