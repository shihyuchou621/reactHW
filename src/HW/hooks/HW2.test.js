import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HW from './HW2';

test('should HW2', () => {
  const { container } = render(<HW />);
  expect(container).toMatchSnapshot();
});

test('list should add 1 item', () => {
  const { container } = render(<HW />);
  const button = container.querySelector('button');
  userEvent.click(button);
  expect(container).toMatchSnapshot();
});

test('list should add 4 item', () => {
  const { container } = render(<HW />);
  const button = container.querySelector('button');
  userEvent.click(button);
  userEvent.click(button);
  userEvent.click(button);
  userEvent.click(button);
  expect(container).toMatchSnapshot();
});
