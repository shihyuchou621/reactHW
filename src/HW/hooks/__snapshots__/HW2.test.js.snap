// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list should add 1 item 1`] = `
<div>
  <div>
    <button>
      submit
    </button>
    <ul>
      <li>
        1
      </li>
    </ul>
  </div>
</div>
`;

exports[`list should add 4 item 1`] = `
<div>
  <div>
    <button>
      submit
    </button>
    <ul>
      <li>
        4
      </li>
      <li>
        3
      </li>
      <li>
        2
      </li>
      <li>
        1
      </li>
    </ul>
  </div>
</div>
`;

exports[`should HW2 1`] = `
<div>
  <div>
    <button>
      submit
    </button>
    <ul />
  </div>
</div>
`;
